src/goto-programs/goto_trace.h

```diff
@@ -138,6 +136,22 @@ class goto_trace_stept
     full_lhs_value.make_nil();
     cond_expr.make_nil();
   }
+
+  goto_programt::const_targett get_pc() const { return pc; }
+
+  /// We don't have a way of enforcing this, but \p pc_ must *not* be a
+  /// singular iterator!
+  void set_pc(goto_programt::const_targett pc_)
+  {
+    pc = std::move(pc_);
+    is_pc_set = true;
+  }
+
+  bool get_is_pc_set() const { return is_pc_set; }
```

Comment:
I disagree. Setting it in the constructor doesn't solve anything - the user could still do:
```c++
goto_trace_stept my_step{goto_programt::const_targett()};
```
Then we're back where we started, with a default-constructed `pc`, and with a less usable class too.

I think the real problem here is using an iterator rather than a (smart) pointer which is nullable.